### DISCUSSION TITLE
Automatic PCAP Import

### DISCUSSION THREAD

**Discussion by 0xShinRai** [Security-Onion-Solutions/securityonion#5684] [Unsupported Versions] [answered]
Hey there!

I was wondering if there is a folder or specific way to automate the ingestion of PCAPs that are generated on target servers.

The environment of our SO install does not currently permit using a network sniffing interface and SPAN/Mirroring. I have been experimenting with generating PCAPs on Servers with tools like dumpcap and somehow sending them to SO (with `socat` in this example), but im having some trouble implementing a method of repeated ingestion of new files and i didn't find any resources online.

Does someone have experience/resources in doing what i am currently trying to achieve? (Generate PCAPs on Servers and Import them into SO automatically)

Thanks in advance and great work with SO, we are loving it so far!

Shin

**Comment by dougburks** [comment, marked as answer]
Seems like you could do something like the following:
- create an `import` directory
- send pcap files to the `import` directory
- create a cron job to run on a periodic basis and run the `import` pcaps through `so-import-pcap` and then delete them

**Reply by 0xShinRai** [reply]
Thank you for the quick answer. I thought about doing what you are suggesting. Is there an ending to a pcap file? Im asking this question because i can imagine running into trouble when the crontab triggers on a pcap that is not fully recieved yet and is still being written to. Does so-import-pcap know if a pcap is "incomplete" or is still being written to by another process?

**Reply by dougburks** [reply]
`so-import-pcap` does check for invalid pcap files:
https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/common/tools/sbin/so-import-pcap#L129-L134

So you might try it out and see if it works for your needs.

**Reply by 0xShinRai** [reply]
Thank you for your help! I have since found out that so-import-pcap actually gives the pcap a unique identifier and refuses to import the same pcap twice. This makes it very easy to write a simple for loop that just deletes the pcaps that have already been imported.

Thanks for the help, i really appreciate it.
